magnetic/Glk/glk.c

```diff
@@ -6123,7 +6180,7 @@ glk_main (void)
 /*---------------------------------------------------------------------*/
 /*  Glk linkage relevant only to the UNIX platform                     */
 /*---------------------------------------------------------------------*/
-#ifdef __unix
+#ifdef TRUE
```

Comment:
Let's change this to `#if 1` since that's universally guaranteed to evaluate to true (in some future world we might update things to change `TRUE` to the standard `true` and then this would stop building: yes, that is a ridiculous scenario to assume, but still....)

And, with that said, in the future we'll want a way to select which Glk startup to use.  Assuming Unix is totally fine in general, because basically everything uses it.  The outlier is Windows Glk, and I suspect most interpreters we'll add here don't support it anyway; but, still, to be proper, eventually we'll want this to be a macro that's set by whoever is using the interpreters, e.g. `cc -DINTGLK_UNIX` or `cc -DINTGLK_WIN` or whatever they wind up being called.

Reply:
I was not sure at all about this particular code.
Is then better to change it to `TARGET_OS_UNIX` like the [next OS conditional](https://github.com/cspiegel/interpreters/blob/70bdb286b98350b9fe20bacc42de0c963070cc16/magnetic/Glk/glk.c#L6239) chunk?

Reply:
That would require the user to define `TARGET_OS_UNIX` when compiling.  That's not necessarily a bad thing: I do the same thing in Bocfel, where you're required to choose a Glk startup type.

The slightly shaky part is that a Unix startup can be used outside of Unix: GlkDOS uses one, as would remglk on Windows, for example. So the name is misleading.  The original code here basically assumes that Unix platforms require a Unix startup, and others don't.  But this is old enough that MacGlk was relevant when it was written, so it can be excused for that assumption.

Ok, so I guess I'm fine with switching to `TARGET_OS_UNIX` and just saying, for Magnetic Scrolls, you have to define your Glk type via `TARGET_OS_UNIX` or `TARGET_OS_MAC`.  That's better than simply assuming a Unix Glk, or assuming a Unix platform is the only place such a Glk startup is used. It doesn't support other Glk startups, but never did, so no big deal. Adding Windows Glk support down the line would be a nice feature to have, though.

This treatment really should happen for Level9 and Agility too, since they have very similar startups--no surprise, given that they're all Glk ports done by Simon Baldwin.  That can happen later, but I just want to make sure I don't forget to point it out.